launch: #起動パラメータ
  ros__parameters:
    slcan: false
    scan: false
    tcp_endpoint: true
    trajectory_planner: true

/**:  #ワイルドカード(ここのパラメータは全ノードから読める: <名前に注意>)
  ros__parameters:
    initial_pose: [-5.45, 0.0, 0.0]  #初期位置[x,y,yaw]

    # 自動の並進  速度[m/s],加速度[m/s^2],加減速度[-m/s^2],躍度[m/s^3]
    linear_max_vel: 6.0
    linear_max_acc: 9.0
    linear_max_dec: 9.0
    linear_max_jer: 0.0
    # 回転  速度[deg/s],加速度[deg/s^2],加減速度[-deg/s^2],躍度[deg/s^3]
    angular_max_vel: 90.0
    angular_max_acc: 180.0
    angular_max_dec: 180.0
    angular_max_jer: 0.0

controller_interface_node:
  ros__parameters:
    # 手動の並進　速度[m/s],加速度[m/s^2],加減速度[-m/s^2]
    linear_max_vel: 3.0
    linear_max_acc: 6.0
    linear_max_dec: 6.0
    # 手動の回転  速度[deg/s],加速度[deg/s^2],加減速度[-deg/s^2]
    angular_max_vel: 90.0
    angular_max_acc: 180.0
    angular_max_dec: 180.0
    #手動の上物　速度[deg/s],加速度[deg/s^2],加減速度[-deg/s^2] RC23−ERのdevelopブランチの348行目を参考
    injection_max_vel: 72.0
    injection_max_acc: 40.0
    injection_max_dec: 40.0

    heartbeat_ms: 100 #ハートビート周期[ms]

    #リスタート、緊急、足回りの手自動、上物の手自動、手動の射角、udp_portのデフォルト値
    defalt_restart_flag: false
    defalt_emergency_flag: true
    defalt_wheel_autonomous_flag: true
    defalt_injection_autonomous_flag: true
    defalt_injection_m0_flag: true
    defalt_pitch: 45.0
    udp_port_ER_main: 50000
    udp_port_ER_sub: 51000
    udp_port_RR: 52000

# controller_common_proces_node:
#   ros__parameters:
#     pole_btn_ms: 100

spline_pid_node:
  ros__parameters:
    interval_ms: 5     #サイクル周期[ms]
    curvature_attenuation_rate: 0.0 #曲率の速度の減衰>=0 (0だと減衰なし)
    linear_planner_vel_limit_gain: 0.6  #速度計画機の速度上限値にかけるゲイン

    linear_planner_gain: 1.3 #並進速度計画機ゲイン
    linear_pos_gain: 3.5  #並進位置ズレのゲイン
    linear_pos_integral_gain: 0.01  #並進位置ズレの積分ゲイン

    angular_planner_gain: 1.7 #回転速度計画機ゲイン
    angular_pos_gain: 2.5 #回転位置ズレのゲイン
    angular_pos_integral_gain: 0.01  #回転位置ズレの積分ゲイン

    linear_pos_tolerance: 0.05  #並進到達位置の許容誤差[m]
    angular_pos_tolerance: 1.0  #回転到達位置の許容誤差[deg]

spline_trajectories_node:
  ros__parameters:
    resolution: 0.01  #起動点の分解能[m]


mcl_2d_node:
  ros__parameters:
    num_of_particle: 50     #粒子数
    odom_convariance:
      param1: 0.0   #回転成分から生まれる回転成分のズレ
      param2: 0.0   #並進成分から生まれる回転成分のズレ
      param3: 0.0   #並進成分から生まれる並進成分のズレ
      param4: 0.0   #回転成分から生まれる並進成分のズレ
      param5: 0.01   #Xのズレ
      param6: 0.01   #Yのズレ
    tf_laser2robot: [0.4655,0.0,0.0,  3.14159,0.0,0.0]  #ロボット->lidarのTF[x,y,z,r,p,y]


injection_param_calculator_node:
  ros__parameters:
    ring_weight: 0.10  #リングの質量
    gravitational_accelerastion: 9.80665  #重力加速度
    air_resistance_coefficient: 0.005321687  #空気抵抗係数
    injection_length: 0.41  #射出機構の長さ
    foundation_hight: 0.2636  #射出機構の地面からの高さ
    calculat_first_velocity: 10.0  #初速度の初期値
    velocity_lim_max: 15.0 #最大初速度
    angle_choice: 45.0  #リングが入りやすい角度
    angle_bounds: 40.0  #射角の境界
    max_loop: 100  #ニュートン法の最大繰り返し数
    pitch_limit: [15.0,45.0]  #射角の最小最大
    yow_limit_m1: [-90.0,0.0] #左の射出機構の方向角の最小最大
    yow_limit_m0: [0.0,90.0] #右の射出機構の方向角の最小最大

injection_interface_node:
  ros__parameters:
    tf_injection2robot_m0: [0.0, 0.0] #ロボット->射出機構0のTF[x,y]
    tf_injection2robot_m1: [0.0, 0.0] #ロボット->射出機構1のTF[x,y]


RANSAC_localization:
  ros__parameters:
    plot_mode: true
    laser_weight: 1.0 #lidarの重み
    odom_weight_liner: 1.0e-2
    odom_weight_angler: 1.0e-2
    voxel_size: 0.05
    trial_num: 40
    inlier_dist_threshold: 0.01
    inlier_length_threshold: 1.3
    tf_laser2robot: [0.4655,0.0,0.0,  3.14159,0.0,0.0]  #ロボット->lidarのTF[x,y,z,r,p,y]
